### PR TITLE
Use volume as Docker storage location

### PR DESCRIPTION
**Pull request by agners**
## Proposed change

Use volume as Docker storage location. This allows to use `overlay2` as file system driver and improves performance significantly.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:
- Link to cli pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast supervisor tests`)
- [ ] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
     "visualstudioexptteam.vscodeintellicode",
     "esbenp.prettier-vscode"
   ],
+  "mounts": [ "type=volume,target=/var/lib/docker" ],
```
Comment (ludeeus) [inline .devcontainer/devcontainer.json:13]: Is this working when you use "Clone in volume"?

Comment (agners) [reply .devcontainer/devcontainer.json:13]: What is "Clone in volume"? What is the use case exactly?

Comment (ludeeus) [reply .devcontainer/devcontainer.json:13]: That's one of the 2 ways of opening a devcontainer.
@pvizeli uses that way I think

Comment (agners) [reply .devcontainer/devcontainer.json:13]: I guess its this option?

image

Cloned my branch using that feature, and it seems to work. I think that feature conflicts with the `workspaceMount` option, but it seems fine with `mounts`.

See: https://code.visualstudio.com/docs/remote/devcontainerjson-reference#_devcontainerjson-properties

Comment (pvizeli) [reply .devcontainer/devcontainer.json:13]: I have to check if that work on windows, because I use all time this clone function

Comment (agners) [reply .devcontainer/devcontainer.json:13]: @pvizeli so this seems to work for @mdegat01 on Apple M1. Have you had a chance to check it on your end?

Comment (mdegat01) [reply .devcontainer/devcontainer.json:13]: Yea this tweak was huge. "Run supervisor" task was timing out (30 seconds) and had to be re-run then taking a good bit to get through the rest of the script and startup before this change. After I think the entire task took maybe 30 seconds.

Comment (pvizeli) [reply .devcontainer/devcontainer.json:13]: Still on my list

**Review by mdegat01 — APPROVED** [review]
I have been making this change locally and using it without issue (and major performance benefits) for a few weeks now. I can say it definitely makes things a lot faster on my M1 mac.

**Comment by pvizeli** [comment]
I will test it or revert it

**Pull request merged**